### PR TITLE
[agent & server] modify mqtt request type format. fix agent http2tls not parse and server status incorrect

### PR DESCRIPTION
**Pull request by qq906907952**
### This PR is for:
- Server
- Agent

### Fixes
Bug 19105 : agent not parse http2 tls.

Bug 19101: server status code incorrect

#### Steps to reproduce the bug
- Bug 19105: use golang send http2/grpc request
- Bug 19101: server set the incorrect status code in request log
#### Changes to fix the bug
- Bug 19105 fix: ebpf_collector set http2tls(23) bitmap when receive http2tls packet. add  http2tls parse.
- Bug 19101 fix: add msg type decision (req/resp/session) to set status code.
#### Affected branches
- main
#### Checklist
-

### CODE REVIEW

File: agent/src/ebpf_collector/ebpf_collector.rs [hunk 1]
```diff
@@ -462,6 +462,10 @@ impl FlowItem {
             protocol_bitmap |= 1 << u8::from(L7Protocol::Http1TLS);
             protocol_bitmap &= !(1 << u8::from(L7Protocol::Http1));
         }
+        if packet.l7_protocol_from_ebpf == L7Protocol::Http2TLS {
```
Comment (rvql) [inline agent/src/ebpf_collector/ebpf_collector.rs:465]: 和461合并，用match

File: agent/src/flow_generator/protocol_logs/http.rs [hunk 2]
```diff
@@ -111,6 +111,9 @@ impl HttpInfo {
 impl From<HttpInfo> for L7ProtocolSendLog {
     fn from(f: HttpInfo) -> Self {
         let mut log = L7ProtocolSendLog {
+            // http2 可能没有:content-length头. 默认长度设置-1 防止server判断为0 
+            req_len: -1,
```
Comment (rvql) [inline agent/src/flow_generator/protocol_logs/http.rs:115]: 这个是否要写到default里？这个字段是否用`Option<usize>`更好？

Comment (qq906907952) [reply agent/src/flow_generator/protocol_logs/http.rs:115]: 改成opion吧

Comment (qq906907952) [reply agent/src/flow_generator/protocol_logs/http.rs]: protocol log的req_size resp_size status_code 都改成了option,转换的时候,统一在L7ProtocolSendLog 把option改成-1

File: agent/src/flow_generator/protocol_logs/pb_adapter.rs [hunk 3]
```diff
@@ -66,15 +66,27 @@ pub struct L7ProtocolSendLog {
 
 impl L7ProtocolSendLog {
     pub fn fill_app_proto_log(self, log: &mut flow_log::AppProtoLogsData) {
-        log.req_len = self.req_len;
-        log.resp_len = self.resp_len;
+        let req_len = if let Some(len) = self.req_len{
+            len as i32
+        }else{
```
Comment (rvql) [inline agent/src/flow_generator/protocol_logs/pb_adapter.rs:71]: 注意格式，`} else {`

File: agent/src/flow_generator/protocol_logs/pb_adapter.rs [hunk 4]
```diff
@@ -66,15 +66,27 @@ pub struct L7ProtocolSendLog {
 
 impl L7ProtocolSendLog {
     pub fn fill_app_proto_log(self, log: &mut flow_log::AppProtoLogsData) {
-        log.req_len = self.req_len;
-        log.resp_len = self.resp_len;
+        let req_len = if let Some(len) = self.req_len{
```
Comment (rvql) [inline agent/src/flow_generator/protocol_logs/pb_adapter.rs:69]: 花括号前应有空格

**Review by rvql — APPROVED** [review]

**Pull request merged**